proposals/lip-0048.md

```diff
@@ -25,15 +25,59 @@ This LIP defines the fee system in a modular way, as currently [used in the Lisk
+```python
+Fee.payFee(USER_ACCOUNT_INITIALIZATION_FEE)
+```
+In addition, developers may decide to call the `payFee` method for transactions that require extensive computation, e.g., when verifying an inclusion proof, or if any other resource is extensively used during transaction execution.
+
+### Fee Processing
+
+We refer to the sum of the minimum fee and the additional fees processed by the `payFee` method as the "consumed" part of the fee. The remainder of the fee, called the "available" part of the fee, is transferred to the validator as a reward. In this way, validators are assumed to prioritize transactions with a higher available fee in case of network congestion. [After the transaction is executed](#after-command-execution), consumed fee is either burned or transferred to a dedicated pool address. 
+
+#### Fee Pool
+
+The option to transfer the consumed fees to a dedicated pool address `ADDRESS_FEE_POOL` could be beneficial for sidechains, as fees collected this way can be used as an incentive for validators and other protocol participants for various purposes: Funding a community pool for on-chain governance spending, sharing funds proportionally among validators per round, incentivizing ecosystem developers, supporting chain maintenance and improvement, rewarding users' staking and voting activities, etc.
+
+In the Lisk mainchain, the consumed fees are always burned. This reduces the total supply, creating a deflationary pressure that works against inflation.
+
+### Cross-chain Message Fees
+
+[Cross-chain messages][lip-0049#ccmschema] (CCMs) facilitate interoperability by enabling cross-chain transfer of data. Each CCM contains a *message fee*, intended to cover the cost of transaction processing on the receiving chain. 
+
+Since managing message fees requires checking and updating the escrow balances for the message fee token, this task is assigned to the Token module. In particular, paying for message fees is called for in the Interoperability module, whose [`sendInternal`][lip-0045#sendinternal] method uses the [`payMessageFee`][lip-0051#paymessagefee] of the Token module.
+
+Note that, for protocol reasons (see, e.g., [`bounce`][lip-0045#bounce] and [`terminateChain`][lip-0045#terminatechain] methods), Interoperability module allows for message fees to be zero. Consequently, setting a minimum message fee should be delegated to each module that provides its own cross-chain commands. 
+
+Similar to transaction fees, the consumed part of the message fee can either be burned or transferred to a fee pool, depending on the chain configuration. The available part of the message fee is left to the relayer in order to cover the cost of CCM execution and to incentivize its participation in CCM relaying.
```

Comment:
Highlight that here consumed fee is only the so-called "additional fees", since there is no minFee.